integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/KafkaWriterVisitor.java

```diff
@@ -0,0 +1,44 @@
+package io.openlineage.spark.agent.lifecycle.plan;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.spark.agent.util.PlanUtils;
+import io.openlineage.spark.agent.util.ScalaConversionUtils;
+import java.util.Collections;
+import java.util.List;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.execution.datasources.SaveIntoDataSourceCommand;
+import org.apache.spark.sql.kafka010.KafkaSourceProvider;
+
+/**
+ * {@link LogicalPlan} visitor that matches an {@link KafkaSourceProvider} and extracts the output
+ * {@link OpenLineage.Dataset} being written.
+ * Gets the first host from 'kafka.bootstrap.servers' option and creates namespace
+ * Extracts a 'topic' option and put it as dataset name
+ *
+ * Write to Kafka using a topic specified in the data is not supported.
+ */
+public class KafkaWriterVisitor
+    extends QueryPlanVisitor<SaveIntoDataSourceCommand, OpenLineage.Dataset> {
+
+  @Override
+  public boolean isDefinedAt(LogicalPlan x) {
+    return x instanceof SaveIntoDataSourceCommand
+        && ((SaveIntoDataSourceCommand) x).dataSource() instanceof KafkaSourceProvider;
+  }
+
+  @Override
+  public List<OpenLineage.Dataset> apply(LogicalPlan x) {
+    SaveIntoDataSourceCommand saveIntoDataSourceCommand = (SaveIntoDataSourceCommand) x;
+
+    String servers =
+        ScalaConversionUtils.asJavaOptional(
+                saveIntoDataSourceCommand.options().get("kafka.bootstrap.servers"))
+            .orElse("");
+    String namespace = "kafka://" + servers.split(",")[0];
+    String topic =
+        ScalaConversionUtils.asJavaOptional(saveIntoDataSourceCommand.options().get("topic"))
+            .orElse("");
```

Comment:
but you can write to a new topic as well. We can try to get all topics from kafka via AdminClient but it would require to support kafka authentication. Maybe we can mark it as unsupported feature?

Reply:
The new topic issue is a good call out. There's no good way to support that AFAIK.